### ISSUE TITLE
Is is possible to let SourceFileInfo have swc's `Program` rather than `Module`?

### ISSUE THREAD

**Issue by magurotuna** [dprint/dprint-swc-ecma-ast-view#4]
I've been re-implementing a few of deno_lint's rules using this crate.
deno_lint uses `swc_ecmascript::ast::Program` as an entry point of linting, therefore it would be nice if this crate also took `Program` instead of `Module`. That is, I'd be happy if `SourceFileInfo` was altered to:

```rust
pub struct SourceFileInfo<'a> {
  pub module: &'a swc_ecmascript::ast::Program, // altered!
  pub source_file: Option<&'a swc_common::SourceFile>,
  pub tokens: Option<&'a Vec<TokenAndSpan>>,
  pub comments: Option<&'a SingleThreadedComments>,
}
```

and the signature of `with_ast_view` to:

```rust
pub fn with_ast_view<'a, T>(
    source_file_info: SourceFileInfo<'_>, 
    with_view: impl FnOnce(&'a Program<'a>) -> T // altered!
) -> T
```

From my taking a look at the code of this crate and how the underlying code generation work, it seems like the above alternation is achievable (although it looks a bit complicated to me :-|)

**Comment by dsherret** [comment]
@magurotuna ah, I see. You'd like to sometimes just analyze a Script. I'll take a look at it (I'd only used parse_module previously and didn't know about parse_program).

**Comment by dsherret** [comment]
Thanks for the suggestion, @magurotuna! This is implemented now in 0.6.0.
